core/commands/files/files.go

```diff
@@ -276,6 +288,27 @@ func getNodeFromPath(ctx context.Context, node *core.IpfsNode, p string) (node.N
 	}
 }
 
+func unlinkNodeIfExists(node *core.IpfsNode, path string) error {
+	dir, name := gopath.Split(path)
+	parent, err := mfs.Lookup(node.FilesRoot, dir)
+	if err != nil {
+		return err
+	}
+
+	pdir, ok := parent.(*mfs.Directory)
+	if !ok {
+		err = fmt.Errorf("No such file or directory: %s", dir)
```

Comment:
Not necessarily true, this could be a file, but not a directory. Probably should have the error message be `"PATH was not a directory"` or something